tests/test_unittest/test_backend.py

```diff
@@ -177,5 +178,33 @@ def test_device_sync_option_registered(backend_name):
     assert backend_opts[option_name]["type"] is bool
 
 
+@pytest.mark.parametrize("rebuild", (True, False))
+@pytest.mark.parametrize("backend_name", CPU_BACKENDS)
+@pytest.mark.parametrize("mode", (2,))
+def test_toolchain_profiling(backend_name: str, mode: int, rebuild: bool):
+    build_info: Dict[str, Any] = {}
+    builder = (
+        StencilBuilder(stencil_def)
+        .with_backend(backend_name)
+        .with_externals({"MODE": mode})
+        .with_options(
+            name=stencil_def.__name__,
+            module=stencil_def.__module__,
+            rebuild=rebuild,
+            build_info=build_info,
+        )
+    )
+    stencil_class = builder.build()
+    assert stencil_class is not None
+    if rebuild:
+        assert build_info.get("parse_time", 0.0) > 0.0
+        assert build_info.get("module_time", 0.0) > 0.0
+        if backend_name.startswith("gt") and not backend_name.endswith("numpy"):
+            assert build_info.get("codegen_time", 0.0) > 0.0
+            assert build_info.get("build_time", 0.0) > 0.0
+    else:
+        assert build_info.get("load_time", 0.0) > 0.0
```

Comment:
Just a question here: why is it needed to access `build_info` with a default value instead of just `build_info["whatever_time"]`? If I understand this correctly, the accessed items should always exist in the dict, right?